ckanext/stadtzhtheme/commands.py

```diff
@@ -125,3 +155,70 @@ def _get_datastore_table_page(self, context, offset=0):
         has_next_page = (len(result['records']) > 0)
 
         return (resource_id_list, has_next_page)
+
+    def _cleanup_storage_files(self, context, resource_path):
+        files_to_delete = []
+        files_to_keep = []
+        for root, dirs, files in os.walk(resource_path, topdown=True):
+            if files:
+                for file in files:
+                    resource_id = ''.join(root.split('/')[-2:]) + file
+                    file_path = os.path.join(root, file)
+                    try:
+                        logic.check_access('resource_show', context)
```

Comment:
We don't have to check this for every file, as the context doesn't change. We can just check it at the start of the method.